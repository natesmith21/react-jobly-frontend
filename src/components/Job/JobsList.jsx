import React, {useEffect, useState} from "react";
import JoblyApi from "../../api";
import JobCard from "./JobCard";
import SearchBar from "../SearchBar";


const JobsList = () => {
    const [jobs, setJobs] = useState([]);

    useEffect(() => {
        async function getJobs() {
            let postions = await JoblyApi.getJobs()
            setJobs(postions)
        }
        getJobs();
    }, []);

    const search = async (q) => {
        let jobs = await JoblyApi.getJobs(q);
        setJobs(jobs)
    }

    return (
        <section>
            <SearchBar searchFor={search} />
            <div className="col-md-10 offset-md-1" >
                {jobs.map(j => <JobCard key={j.id} job={j} />)}
            </div>
        </section>
    )
    
}

export default JobsList;