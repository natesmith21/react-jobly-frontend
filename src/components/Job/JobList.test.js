import React from "react";
import { render } from "@testing-library/react";
import JobsList from "./JobsList";

it("renders without crashing", function() {
  render(<JobsList />);
});

it("matches snapshot with no jobs", function() {
  const { asFragment } = render(<JobsList />);
  expect(asFragment()).toMatchSnapshot();
});
