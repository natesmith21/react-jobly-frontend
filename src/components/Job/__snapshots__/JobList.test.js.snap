// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no jobs 1`] = `
<DocumentFragment>
  <section>
    <form
      class="col-md-8 offset-md-2 searchBar"
    >
      <label
        class="form-label"
        for="q"
      />
      <input
        class="form-control"
        id="q"
        name="search"
        placeholder="search"
        type="text"
        value=""
      />
      <button
        class="btn btn-secondary"
      >
        Submit
      </button>
    </form>
    <div
      class="col-md-10 offset-md-1"
    />
  </section>
</DocumentFragment>
`;
